### PR TITLE
Fix: Correct JSON Schema additionalProperties for get_schema MCP Tool (#143)

### LINKED ISSUE #143: [HELP WANTED] Bug: MCP Server schema validation error with Gemini CLI
---
name: Help Wanted
about: Request community help on a specific task or feature
title: '[HELP WANTED] Bug: MCP Server schema validation error with Gemini CLI'
labels: 'help wanted', 'bug'
assignees: ''
---

## What We Need Help With
We need help investigating and fixing a schema validation error that occurs when connecting the Intugle MCP server (`intugle-mcp`) to the Gemini CLI. The error indicates a mismatch in the expected type for `additionalProperties` in the tool's output schema.

## Background & Context
The Model Context Protocol (MCP) server allows AI agents to interact with Intugle's semantic layer. However, when users try to connect the `intugle-mcp` server to the Gemini CLI, the connection fails during tool discovery due to a JSON schema validation error.

## Current State
When the Gemini CLI attempts to discover tools from the Intugle MCP server, it throws the following error:

```json
Error discovering tools from semantic_model: [
    {
      "code": "invalid_type",
      "expected": "boolean",
      "received": "object",
      "path": [
        "tools",
        1,
        "outputSchema",
        "additionalProperties"
      ],
      "message": "Expected boolean, received object"
    }
]
```

This suggests that for one of the tools (specifically the second one, index 1), the `outputSchema` defines `additionalProperties` as an object, but the consumer (Gemini CLI) expects a boolean.

## Desired Outcome
The `intugle-mcp` server should successfully connect to the Gemini CLI without throwing schema validation errors. The tool definitions should generate schemas that are compatible with the strict validation expected by the client.

## Scope of Work
- [ ] Reproduce the issue using the steps provided below.
- [ ] Identify which specific tool in the MCP server implementation is causing the schema issue.
- [ ] Modify the Pydantic model or schema generation logic to ensure `additionalProperties` is handled correctly (likely by setting it to `False` or removing it if not needed, or ensuring the client can handle it).
- [ ] Verify the fix by successfully connecting to the Gemini CLI.

### Relevant Files/Modules
- `src/intugle/mcp/` (Likely contains the server implementation and tool definitions)
- `src/intugle/mcp/server.py` (or similar entry point)

### Key Concepts/Technologies
- **Model Context Protocol (MCP)**
- **Pydantic**: Used for defining data models and generating JSON schemas.
- **JSON Schema**: Understanding `additionalProperties` behavior.

### Related Documentation
- [Pydantic JSON Schema generation](https://docs.pydantic.dev/latest/usage/json_schema/)

## Steps to Reproduce
1. Install the library:
   ```bash
   pip install intugle
   ```
2. Start the MCP server:
   ```bash
   intugle-mcp
   ```
3. Configure the Gemini CLI (or create a `.gemini/settings.json`) with the following server configuration:
   ```json
   "mcpServers": {
     "semantic_model": {
       "httpUrl": "http://localhost:8080/semantic_layer/mcp",
       "trust": true
     }
   }
   ```
4. Start the Gemini CLI. The error should appear immediately upon initialization.

## Suggested Approach
1. Inspect the `src/intugle/mcp/` directory to find where tools are defined.
2. Look for Pydantic models used as return types for the tools.
3. Check if any model config has `extra = 'allow'` or similar settings that might generate an object for `additionalProperties`.

## Skills Needed
- [x] Python development
- [ ] Machine Learning (XGBoost, scikit-learn)
- [ ] LLM/GenAI integration (LangChain, OpenAI)
- [ ] Data engineering
- [ ] SQL and database knowledge
- [ ] Vector databases (Qdrant)
- [ ] Databricks/Snowflake experience
- [ ] Frontend/UI (Streamlit)
- [ ] Documentation writing
- [ ] Testing (pytest)
- [x] Other: Pydantic, JSON Schema

## Getting Started
1. Fork the repository
2. Set up your development environment (see [CONTRIBUTING.md](https://github.com/Intugle/data-tools/blob/main/CONTRIBUTING.md))
3. Reproduce the error locally.
4. Apply the fix in the Pydantic models or server configuration.

## Testing Requirements
- [ ] Manual testing: Verify the error is gone when restarting the Gemini CLI.
- [ ] Unit tests: Ensure the generated schema for the tools is valid.

## Questions & Support
- Ask questions in the comments below
- Join our [Discord](https://discord.gg/NqR9tNWVTm) for real-time help

---
**We appreciate your interest in contributing to Intugle! 🙏**

### PR DESCRIPTION
**Pull request by ice-user**
## Description
This PR fixes the MCP schema validation error encountered when connecting the Intugle MCP server to the Gemini CLI.  
The issue occurred because the `get_schema` tool returned a type annotated as `dict[str, str]`, causing FastMCP to generate:

```json
"additionalProperties": { "type": "string" }
```

but gemini-cli expects additionalProperties to be a boolean
The fix changes the return type to dict, which correctly generates:
```json
"additionalProperties": true
```

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] 🎨 Code style update (formatting, renaming)
- [ ] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [ ] ✅ Test update
- [ ] 🔧 Configuration change
- [ ] 🏗️ Infrastructure/build change

## Related Issue(s)

Fixes #143 

## Changes Made

-Updated the return type of the get_schema tool in mcp/semantic_layer/router from dict[str, str] to dict.
-Regenerated MCP tool schema via FastMCP.
-Confirmed Gemini CLI successfully connects and executes tool discovery.

### Test Configuration
- **Python Version**: 3.11.7
- **OS**: Arch linux
- **LLM Provider** (if applicable): Gemini CLI (MCP client)

### Test Cases
- [x] Unit tests pass locally
- [ ] Manual testing completed
- [ ] Tested with sample datasets

### Test Commands
```bash
# Commands to run tests
pytest tests/
```
Run MCP server:
intugle-mcp

## Screenshots/Examples

before fix:
image

after fix:
image

image

## Checklist

- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or linter errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published
- [ ] I have updated the relevant notebooks (if applicable)
- [ ] I have checked my code and corrected any misspellings

## Documentation Updates

- [ ] README.md updated
- [ ] Docstrings added/updated
- [ ] Documentation site updated (if needed)
- [ ] Notebook examples updated (if applicable)
- [ ] CHANGELOG updated (if applicable)

## Breaking Changes

- [x] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)

## Performance Impact

- [ ] Performance benchmarks run
- [x] No significant performance impact
- [ ] Performance improvement: 
- [ ] Performance regression: 

## Additional Context

No special deployment steps required
## Deployment Notes

N/A

---

### CODE REVIEW

**Comment by raphael-intugle** [comment]
Looks like this was a bug in Gemini CLI which has been resolved in the newest version. Closing
